extensions/wikia/Email/Controller/WatchedPageController.class.php

```diff
@@ -122,38 +120,178 @@ private function getDetails() {
 	/**
 	 * @return String
 	 */
-	private function getCompareChangesLabel() {
+	protected function getButtonText() {
 		return wfMessage( 'emailext-watchedpage-diff-button-text' )->inLanguage( $this->targetLang )->text();
 	}
 
 	/**
 	 * @return String
 	 */
-	private function getCompareChangesLink() {
+	protected function getButtonLink() {
 		return $this->title->getFullUrl( [
-			'diff' => $this->currentRevId,
-			'oldid' => $this->previousRevId
+			'diff' => $this->currentRevId
 		] );
 	}
 
 	/**
 	 * @return String
 	 */
-	private function getArticleLinkText() {
+	protected function getArticleLinkText() {
 		return wfMessage( 'emailext-watchedpage-article-link-text',
 			$this->title->getFullURL( [
 				'diff' => 0,
 				'oldid' => $this->previousRevId
 			] ),
-			$this->title->getPrefixedText() )->inLanguage( $this->targetLang )->parse();
+			$this->title->getPrefixedText()
+		)->inLanguage( $this->targetLang )->parse();
+	}
+
+	/**
+	 * @return String
+	 */
+	protected function getAllChangesText() {
+		return wfMessage( 'emailext-watchedpage-view-all-changes',
+			$this->title->getFullURL( [
+				'action' => 'history'
+			] ),
+			$this->title->getPrefixedText()
+		)->inLanguage( $this->targetLang )->parse();
+	}
+
+	/**
+	 * @return Array
+	 */
+	protected function getContentFooterMessages() {
+		return [
+			$this->getArticleLinkText(),
+			$this->getAllChangesText(),
+		];
+	}
+}
+
+class WatchedPageEditedController extends WatchedPageController {
+	/**
+	 * @return String
+	 */
+	protected function getSubjectMessageKey() {
+		return 'emailext-watchedpage-article-edited-subject';
+	}
+
+	/**
+	 * @return String
+	 */
+	protected function getSummaryMessageKey() {
+		return 'emailext-watchedpage-article-edited';
+	}
+}
+
+class WatchedPageProtectedController extends WatchedPageController {
+	/**
+	 * @return String
+	 */
+	protected function getSubjectMessageKey() {
+		return 'emailext-watchedpage-article-protected-subject';
+	}
+
+	/**
+	 * @return String
+	 */
+	protected function getSummaryMessageKey() {
+		return 'emailext-watchedpage-article-protected';
+	}
+}
+
+class WatchedPageUnprotectedController extends WatchedPageController {
+	/**
+	 * @return String
+	 */
+	protected function getSubjectMessageKey() {
+		return 'emailext-watchedpage-article-unprotected-subject';
+	}
+
+	/**
+	 * @return String
+	 */
+	protected function getSummaryMessageKey() {
+		return 'emailext-watchedpage-article-unprotected';
+	}
+}
+
+class WatchedPageDeletedController extends WatchedPageController {
+	/**
+	 * @return String
+	 */
+	protected function getSubjectMessageKey() {
+		return 'emailext-watchedpage-article-deleted-subject';
+	}
+
+	/**
+	 * @return String
+	 */
+	protected function getSummaryMessageKey() {
+		return 'emailext-watchedpage-article-deleted';
+	}
+
+	/**
+	 * @return String
+	 */
+	protected function getButtonText() {
+		return wfMessage( 'emailext-watchedpage-deleted-button-text' )->inLanguage( $this->targetLang )->text();
```

Comment:
you could just leave this method in abstract class, and create only key-getter for message text - just like with getSummaryMessageKey or getSubjectMessageKey